metricbeat/helper/prometheus/metric.go

```diff
@@ -258,6 +267,28 @@ func (o opFilter) Process(field string, value interface{}, labels common.MapStr)
 	return field, value, labels
 }
 
+type opFilterMap struct {
+	label     string
+	filterMap map[string]string
+}
+
+// Called by the Prometheus helper to apply extra options on retrieved metrics
+// Check whether the value of the specified label is allowed and, if yes, return the metric via the specified mapped field
+// Else, if the specified label does not match the filter, return nil
+// This is useful in cases where multiple Metricbeat fields need to be defined per Prometheus metric, based on label values
+func (o opFilterMap) Process(field string, value interface{}, labels common.MapStr) (string, interface{}, common.MapStr) {
```

Comment:
double check if receiver "o" should be an opFilterMap or *opFilterMap. there's reasons to use one over the other. i think it doesn't matter now because this function does not modify "o", but for future-proof and consistency this should be looked at.

Reply:
Should be o opFilterMap See https://github.com/golang/go/wiki/CodeReviewComments#pass-values.